dali/kernels/imgproc/colour_manipulation/brightness_contrast.h

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_COLOUR_MANIPULATION_BRIGHTNESS_CONTRAST_H_
+#define DALI_KERNELS_IMGPROC_COLOUR_MANIPULATION_BRIGHTNESS_CONTRAST_H_
+
+#include "dali/util/ocv.h"
+#include "dali/core/common.h"
+#include "dali/core/geom/box.h"
+#include "dali/core/error_handling.h"
+#include "dali/kernels/kernel.h"
+#include "dali/pipeline/data/types.h"
+
+namespace dali {
+namespace kernels {
+
+namespace brightness_contrast {
+
+struct Roi {
+  Roi() = default;
+
+
+  Roi(int x, int y, int w, int h) : x(x), y(y), w(w), h(h), default_roi(false) {
+    valid_roi = w >= 0 && h >= 0;
+  }
+
+
+  Roi(const Box<3, int> &box) : // NOLINT non-explicit ctor
+          Roi(box.lo[0], box.lo[1], box.extent()[0], box.extent()[1]) {
+  }
+
+
+  int x, y, w, h;
+  bool valid_roi = true;
+  bool default_roi = true;
+};
+
+}  // namespace brightness_contrast
+
+template<typename ComputeBackend, typename InputType, typename OutputType>
+class DLL_PUBLIC BrightnessContrast {
+ private:
+  using StorageBackend = compute_to_storage_t<ComputeBackend>;
+  static constexpr size_t ndims = 3;
+
+ public:
+  DLL_PUBLIC KernelRequirements
+  Setup(KernelContext &context, const InTensor<StorageBackend, InputType, ndims> &image,
+        InputType brightness, InputType contrast, brightness_contrast::Roi roi = {}) {
+    AdjustRoi(roi, image.shape);
+    KernelRequirements req;
+    req.output_shapes = {TensorListShape<DynamicDimensions>({roi_to_shape<ndims>(roi)})};
+    return req;
+  }
+
+
+  // TODO(mszolucha): CHW layout
```

Comment:
I wonder if we need to handle channels at all. For RGB we can treat channel as additional spatial dimension.